action/Run-Maester.ps1

```diff
@@ -130,11 +139,24 @@ PROCESS {
         }
     }
 
+    if ([string]::IsNullOrWhiteSpace($TeamsChannelId) -eq $false -and [string]::IsNullOrWhiteSpace($TeamsTeamId) -eq $false) {
+        $MaesterParameters.Add( 'TeamChannelId', $TeamsChannelId )
+        $MaesterParameters.Add( 'TeamId', $TeamsTeamId )
+        Write-Host "Results will be sent to Teams Team Id: $TeamsTeamId"
+    }
+
```

Comment:
It could be useful here to output a warning if the parameter is used, but an empty or whitespace value is provided. (I think a warning that skips the Teams notification would be better than to fail the entire job.)

You could check this with a combination of `if ($PSBoundParameters.ContainsKey('TeamsChannelId') -or $PSBoundParameters.ContainsKey('TeamsTeamId'))` and `if ([string]::IsNullOrWhiteSpace($TeamsChannelId) -or [string]::IsNullOrWhiteSpace($TeamsTeamId))` to track the warning condition.

Reply:
You could also use parameter validation, but that would have the more direct effect of killing the automated action if valid channel and team ID parameters are not provided, so in this case, I like the approach of validating within the script.